fdb-extensions/src/main/java/com/apple/foundationdb/synchronizedsession/SynchronizedSession.java

```diff
@@ -216,15 +224,48 @@ public static void endAnySession(@Nonnull Transaction tr, @Nonnull Subspace lock
         tr.clear(lockSubspace.range());
     }
 
-    private CompletableFuture<UUID> getLockSessionId(@Nonnull Transaction tr) {
-        return tr.get(lockSessionIdSubspaceKey)
+
+    /**
+     * Check if there is any active session on the given lock subspace, so that a new session would not able to be initialized.
+     * @param tr transaction to use
+     * @param lockSubspace the lock whose active session needs to be checked
+     * @return {@code true} if there is any active session, otherwise {@code false}
+     */
+    public static CompletableFuture<Boolean> checkAnySession(@Nonnull Transaction tr, @Nonnull Subspace lockSubspace) {
```

Comment:
Bit of a nit, but something about "any" doesn't quite seem right here... Maybe `checkActive`?

Reply:
Yeah, I called it `checkAnySession` because it's a "parallel narrative" of the `endAnySession`  methods above, but now I can see how this name can be misleading. 

However, `checkActive` also sound vague to me because it doesn't say it's about the current session or about all sessions of the lock. I would imagine `SynchronizedSession` could be refactored to two classes: `SynchronizedSessionLock` (defined by a lock subspace) and a `SynchronizedSession` (defined by a lock subspace and a session ID). After that, `checkActive` should be a good name on `SynchronizedSessionLock`.

What about `checkExistActiveSession`?

Reply:
Okay, I think that name makes sense. Just a grammar change suggested in a follow up review comment